### ISSUE TITLE
Option to fail Gradle detekt task upon violations

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#63]
An extension point for letting the tool fail when violations are found would be beneficial.

I'd assume it'd just pass the configuration flag downstream to `detekt-cli`.

**Comment by arturbosch** [comment]
I think you want build failures. Add this to your configuration file:

```yaml
build:
  warningThreshold: 5
  failThreshold: 10
  weights:
    complexity: 2
    formatting: 0
    LongParameterList: 1
    comments: 0.5
```

**Comment by vanniktech** [comment]
Awesome, thanks.

How does the weighing work?

**Comment by arturbosch** [comment]
Every non specified smell/rule set gets a multiplier by 1. One finding = 1 weight point. If you have more than 5 points, a warning is shown in the console. More than 10, the build will fail.

0.5 actually does not work :D
